### PR TITLE
fix(headless): handle custom Coveo domains in basic configuration

### PR DESCRIPTION
**Pull request by ThibodeauJF**
### For “platform” clients, any environment (prod,dev,...) and any region (us,eu,...)

SearchAPI Endpoint: 
https://platform.cloud.coveo.com/rest/search/v2?organizationId=myorg

Analytics Endpoint:
https://analytics.cloud.coveo.com/rest/ua/v15/analytics/search?visitor=xxxxxxxxxxx

### For "custom domain" clients (orgname.org.coveo.com)

SearchAPI Endpoint: 
https://myorg.org.coveo.com/rest/search/v2?organizationId=myorg

Analytics Endpoint (what this PR allows):
https://analytics.cloud.coveo.com/rest/ua/v15/analytics/search?visitor=xxxxxxxxxxx

**Potential issues:**
    - Can be a different environment (qa, dev, hippa) (unsure if this applies to custom domains)
    - Can be a different region (eu)

To specify a different analytics endpoint than the basic one, a customer would need to dispatch an additional action after configuration:

```typescript
const apiBaseUrl = 'https://analyticsdev-eu.cloud.coveo.com/rest/ua'; // could also use `analyticsUrl` helper
const action = loadConfigurationActions(engine).updateAnalyticsConfiguration({ apiBaseUrl });
engine.dispatch(action);
```
https://coveord.atlassian.net/browse/JSUI-3308

### CODE REVIEW

File: packages/headless/src/features/configuration/configuration-slice.ts [hunk 1]
```diff
@@ -17,13 +17,28 @@ import {
...
+) {
+  const isCoveoPlatformURL =
+    /^https:\/\/platform(dev|qa|hipaa)?(-)?(eu|au)?\.cloud\.coveo\.com/.test(
+      platformUrl
+    );
+  if (isCoveoPlatformURL) {
+    return platformUrl.replace(/^(https:\/\/)platform/, '$1analytics');
   }
+
+  const isCoveoOrgDomainUrl = new RegExp(
```
Comment (olamothe) [inline packages/headless/src/features/configuration/configuration-slice.ts:32]: You should be able to achieve the same thing by building only one regex (the second one, with the capturing group).

Suggestion: 
```
const isCoveoOrgDomainUrlMatch = platformUrl.match(new RegExp(`^https://(${organizationId}\\.org)\\.coveo.com`))

if(isCoveoOrgDomainUrlMatch) {
   return platformUrl.replace(isCoveoOrgDomainUrlMatch[1], 'analytics.cloud')
}

File: packages/headless/src/api/platform-client.ts [hunk 2]
```diff
@@ -119,6 +120,10 @@ function coveoCloudURL<E extends PlatformEnvironment>(
 export function platformUrl<E extends PlatformEnvironment>(
   options?: URLOptions<E>
 ) {
+  if (options?.orgDomain) {
+    return `https://${options.orgDomain}.org.coveo.com`;
```
Comment (samisayegh) [inline packages/headless/src/api/platform-client.ts:124]: Why is `.org.coveo.com` hard-coded? Wouldn't developers want to control this part of the url?

Comment (ThibodeauJF) [reply packages/headless/src/api/platform-client.ts:124]: Is is for multi-region deployments https://docs.coveo.com/en/2976/coveo-solutions/deployment-regions-and-strategies#configuration

Comment (samisayegh) [reply packages/headless/src/api/platform-client.ts:124]: Got it, thanks. Naming the option something like `multiRegionSubDomain` could be helpful.

Another alternative could be to just accept the full domain (i.e. no option on the platformUrl utility). The reducer can detect whether it is multi-region or not as it is currently, and set the correct analytics endpoint.

**Pull request merged**